Source/ToolCore/JSBind/JSBClass.h

```diff
@@ -82,13 +80,55 @@ class JSBProperty
         }
 
         if (allUpper)
-        {
             return name_.ToLower();
+
+        if (isupper(name_[0]))
+            newName[0] = tolower(name_[0]);
+
+        if (nameLength == 1)
+            return newName;
+
+        char prevChar = name_[0];
+        for (int k = 1; k < nameLength - 1; k++)
+        {
+            char currChar = name_[k];
+            char nextChar = name_[k + 1];
+
+            if ('0' <= nextChar && nextChar <= '9')
+            {
+                newName[k] = tolower(currChar);
+            }
+            else if (isupper(currChar) && isupper(prevChar) && isupper(nextChar))
+            {
+                newName[k] = tolower(currChar);
+            }
+
+            prevChar = currChar;
+        }
+        if (isupper(name_[nameLength - 1]) && isupper(prevChar))
+            newName[nameLength - 1] = tolower(name_[nameLength - 1]);
+
+        //Checks IOS at front
+        if (name_.StartsWith("IOS", false))
+        {
+            newName[0] = tolower(name_[0]);
+            newName[1] = tolower(name_[1]);
+            newName[2] = tolower(name_[2]);
+        }
+
+        int lastLetter = nameLength - 1;
+        int secondLast = nameLength - 2;
+        int thirdLast = nameLength - 3;
+
+        //Check IOS at back
+        if (name_.EndsWith("IOS", false))
+        {
+            newName[lastLetter] = toupper(name_[lastLetter]);
+            newName[secondLast] = toupper(name_[secondLast]);
+            newName[thirdLast] = toupper(name_[thirdLast]);
         }
```

Comment:
This is in @Vlamboljant's fork, so no we don't have it in our's. What suggestions do you have for him on improving it?